src/main/java/com/clickhouse/kafka/connect/sink/db/mapping/Column.java

```diff
@@ -123,7 +123,7 @@ public static Column extractColumn(String name, String valueType, boolean isNull
             String mapValue = val[1].trim();
             return new Column(name, type, false, dispatchPrimitive(mapKey), dispatchPrimitive(mapValue));
         } else if (valueType.startsWith("LowCardinality")) {
-            return extractColumn(name, valueType.substring("LowCardinality".length() + 1, valueType.length() - 1), true);
+            return extractColumn(name, valueType.substring("LowCardinality".length() + 1, valueType.length() - 1), isNull);
```

Comment:
This is the fix. All LowCard were assumed to be nullable while they are not nullable (sometimes)